src/Hint/NumLiteral.hs

```diff
@@ -43,10 +45,16 @@ numLiteralHint _ modu =
   -- not the module so to be safe, look also at `firstDeclComments
   -- modu` (https://gitlab.haskell.org/ghc/ghc/-/merge_requests/9517).
   let exts = union (extensions (modComments modu)) (extensions (firstDeclComments modu)) in
+  -- TODO: there's a subtle bug when the module disables `NumericUnderscores`.
+  -- This seems pathological, though, because who would enable it for their
+  -- project but disable it in specific files?
   if NumericUnderscores `elem` exts then
      concatMap suggestUnderscore . universeBi
   else
      const []
+  where
+    moduleExtensions = union (extensions (modComments modu)) (extensions (firstDeclComments modu))
+    activeExtensions = configuredExtensions modu <> toList moduleExtensions
```

Comment:
`activeExtensions` doesn't seem to be used anywhere?

Reply:
Great catch, thanks! One of the rebases must've broken this (I've fixed conflicts in this PR many times since it was originally opened in late 2022).